core/src/main/scala/kafka/consumer/BaseConsumer.scala

```diff
@@ -91,6 +94,26 @@ class NewShinyConsumer(topic: Option[String], partitionId: Option[Int], offset:
     }
   }
 
+  def seekToRealPositionsBeforeExit() {
+    while (recordIter.hasNext) {
+      val record = recordIter.next()
+      if (!consumedOffsets.exists(tp => tp._1.topic() == record.topic() && tp._1.partition() == record.partition())) {
+        val tp = new TopicPartition(record.topic(), record.partition())
+        consumedOffsets += tp -> record.offset()
+      }
+    }
+    consumedOffsets.foreach { case (tp, offset) => consumer.seek(tp, offset) }
+  }
+
+  def updateOffsetForPartition(msg: BaseConsumerRecord) {
```

Comment:
I don't think we need this. We only care about the unconsumed data which is still in the iterator when we close. Any partitions which do not have any pending data in the iterator will already be at the right position.